ddtrace/internal/datadog/profiling/dd_wrapper/include/uploader.hpp

```diff
@@ -17,16 +20,25 @@ struct DdogProfExporterDeleter
 
 class Uploader
 {
+  private:
+    static inline std::mutex upload_lock{};
+
+  public:
     size_t profile_seq = 0;
     std::string runtime_id;
     std::unique_ptr<ddog_prof_Exporter, DdogProfExporterDeleter> ddog_exporter;
     std::string url;
 
     std::string errmsg;
 
+    static inline ddog_CancellationToken* cancel{ nullptr };
```

Comment:
The token is static?

Reply:
I'm a little nervous about all the static state we're dangling around.  Can make it hard to test